src/msg/async/AsyncConnection.cc

```diff
@@ -1019,6 +1032,10 @@ ssize_t AsyncConnection::_process_connection()
         r = net.reconnect(get_peer_addr(), sd);
         if (r < 0) {
           ldout(async_msgr->cct, 1) << __func__ << " reconnect failed " << dendl;
+          if (r == -ECONNREFUSED) {
+            ldout(async_msgr->cct, 2) << __func__ << " connection refused!" << dendl;
+            center->dispatch_event_external(connection_refused_handler);
```

Comment:
On Thu, Jul 14, 2016 at 10:19:57AM -0700, Haomai Wang wrote:
> > @@ -1019,6 +1032,10 @@ ssize_t AsyncConnection::_process_connection()
> >          r = net.reconnect(get_peer_addr(), sd);
> >          if (r < 0) {
> >            ldout(async_msgr->cct, 1) << __func__ << " reconnect failed " << dendl;
> > +          if (r == -ECONNREFUSED) {
> > +            ldout(async_msgr->cct, 2) << __func__ << " connection refused!" << dendl;
> > +            center->dispatch_event_external(connection_refused_handler);
> 
> async msgr now also use dispatch_queue-> queue_refused !

Updated and rebased.

-- 
Piotr Dałek
branch@predictor.org.pl
http://blog.predictor.org.pl